server/app/repository/AccountRepository.java

```diff
@@ -163,35 +165,37 @@ public CompletionStage<Void> updateApplicant(ApplicantModel applicant) {
         executionContext);
   }
 
-  public void updateClientDob(String newDob, ApplicantModel applicant) {
-    applicant.getApplicantData().setDateOfBirth(newDob);
-    applicant.save();
-  }
-
-  public void updateClientName(
-      String firstName, String middleName, String lastName, ApplicantModel applicant) {
-    applicant.getApplicantData().updateUserName(firstName, middleName, lastName);
-    applicant.save();
-  }
-
-  public void updateClientPhoneNumber(String phoneNumber, ApplicantModel applicant) {
-    applicant.getApplicantData().setPhoneNumber(phoneNumber);
-    applicant.save();
-  }
-
-  public void updateClientTiNote(String tiNote, AccountModel account) {
-    account.setTiNote(tiNote);
-    account.save();
-  }
-
-  public void updateClientEmail(String email, AccountModel account) {
-    if (!Strings.isNullOrEmpty(email)) {
-      if (lookupAccountByEmail(email).isPresent()) {
-        throw new EmailAddressExistsException();
+  public void updateTiClient(
+      AccountModel account,
+      ApplicantModel applicant,
+      String firstName,
+      String middleName,
+      String lastName,
+      String phoneNumber,
+      String tiNote,
+      String email,
+      String newDob) {
+
+    try (Transaction transaction = database.beginTransaction(TxIsolation.SERIALIZABLE)) {
+      transaction.setBatchMode(true);
+      // new email should different from the current email
+      if (!email.equals(account.getEmailAddress())) {
+        // if the new email is not null or null, then it should not be attached to any other
```

Comment:
suggestion (blocking): This comment is confusing as it's currently written. Should it be something like "If the new email is not null, check that it is not attached to any existing account"? I also think the code is clear enough that you could take this comment out.

Reply:
removed @rockycodes